nservicebus/transports/transactions.md

```diff
@@ -45,11 +45,26 @@ NOTE: MSMQ will escalate to a distributed transaction right away since it doesn'
 snippet: TransportTransactionScope
 
 
-#### Consistency guarantees
+#### Atomicity and consistency guarantees
 
-In this mode handlers will execute inside the `TransactionScope` created by the transport. This means that all the data updates are executed as a whole or rolled back as a whole.
+In this mode handlers will execute inside a `TransactionScope` created by the transport. This means that all the data updates and queue operations are all committed or all rolled back.
 
-Distributed transactions do not guarantee atomicity for an outside observer. For example outgoing messages might be dispatched and processed by some other endpoint before changes to the database get committed. However it is guaranteed that eventually all resources will sync up to reflect the outcome of the transaction.
+**A distributed transaction between the queueing system and the persistent storage guarantees _atomic commits_ but does not guarantee _consistency_.**
+
+Consider a message exchange scenario with a saga that models a simple order lifecycle:
+
+1. `OrderSaga` receives a `StartOrder` message
+1. New `OrderSagaData` instance is created and stored in RavenDB.
+1. `OrderSaga` sends `VerifyPayment` message to `PaymentService`.
+1. NServiceBus completes the distributed transaction and the DTC instructs MSMQ and RavenDB resource managers to commit their local transactions.
+1. `StartOrder` message is removed from the input queue and `VerifyPayment` is immediately sent to `PaymentService`.
+1. RavenDB transaction is committed and changes are written to the transaction log.
+1. `PaymentService` receives `VerifyPayment` message and immediately responds with a `CompleteOrder` message to the originating `OrderSaga`.
+1. `OrderSaga` receives the `CompleteOrder` message and attempts to complete the saga.
+1. `OrderSaga` queries RavenDB to find the `OrderSagaData` instance to complete.
+1. RavenDB returns an empty result set and `OrderSaga` fails to complete. This may be caused either by the DTC or the snapshot isolation guarantee provided by the underlying storage engine (Esent).
+
```

Comment:
We've seen it with RavenDB. I'll add a note that it is also possible with lower isolation levels on relational databases.